apps/server/src/controllers/ontimeController.config.ts

```diff
@@ -0,0 +1,12 @@
+import { ensureJsonExtension } from '../utils/ensureJsonExtension.js';
+
+export const sanitizeProjectFilename = (req, res, next) => {
+  const { filename, newFilename } = req.body;
+  const { filename: projectName } = req.params;
+
+  req.body.filename = ensureJsonExtension(filename);
+  req.body.newFilename = ensureJsonExtension(newFilename);
+  req.params.filename = ensureJsonExtension(projectName);
+
+  next();
+};
```

Comment:
question: i am unsure on why the difference between the config and validate files. should we simplify and move this to the ontimeController.validate.ts?
Or perhaps call it a middleware instead?

What do you think?

Reply:
my idea is that validate is only responsible for validation, it doesn't change anything, while this one changes data

I think a middleware is probably a better idea then

Reply:
Sounds good. Just remember we are doing the fastify migration. We don't want to do double work here, it is ok with me to leave this as a function for now and then move it to a middleware in fastify

Reply:
tbh it's already a middleware, just on the config file.

I moved it to the utils folder (like other middleware we have), and we can place it in a proper middleware folder when we do the migration